### ISSUE TITLE
Inheritance-diagram generation fails when a class is derived from collections.namedtuple

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#510]
Autodoc on the following short program fails when built with 1.0.1.  Removing the inheritance-diagram directive makes the failure go away.

```
#!python
r"""                                                                                                                                                        
MyModule docstring starts here                                                                                                                            
.. inheritance-diagram:: mymod                                                                                                                         
"""
import collections
class Foo(collections.namedtuple('Foo_tuple_name', 'bar baz')):
    """                                                                                                                                                     
    A specialization of namedtuple with three attributes.                                                                                                   
    """
    pass
```

Here's the error message:

```
Exception occurred:
  File "/usr/lib/python2.6/site-packages/sphinx/environment.py", line 703, in read_doc
    pickle.dump(doctree, f, pickle.HIGHEST_PROTOCOL)
PicklingError: Can't pickle <class 'pkg1.mymod.Foo_tuple_name'>: attribute lookup pkg1.mymod.Foo_tuple_name failed
The full traceback has been saved in /tmp/sphinx-err-96a0zi.log, if you want to report the issue to the developers.
```

Here's the full traceback:

```
# Sphinx version: 1.0.1
# Docutils version: 0.6 release
# Jinja2 version: 2.2.1
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/sphinx/cmdline.py", line 173, in main
    app.build(force_all, filenames)
  File "/usr/lib/python2.6/site-packages/sphinx/application.py", line 207, in build
    self.builder.build_update()
  File "/usr/lib/python2.6/site-packages/sphinx/builders/__init__.py", line 198, in build_update
    'out of date' &#37; len(to_build))
  File "/usr/lib/python2.6/site-packages/sphinx/builders/__init__.py", line 218, in build
    purple, length):
  File "/usr/lib/python2.6/site-packages/sphinx/builders/__init__.py", line 120, in status_iterator
    for item in iterable:
  File "/usr/lib/python2.6/site-packages/sphinx/environment.py", line 515, in update_generator
    self.read_doc(docname, app=app)
  File "/usr/lib/python2.6/site-packages/sphinx/environment.py", line 703, in read_doc
    pickle.dump(doctree, f, pickle.HIGHEST_PROTOCOL)
PicklingError: Can't pickle <class 'pkg1.mymod.Foo_tuple_name'>: attribute lookup pkg1.mymod.Foo_tuple_name failed
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/510
- Originally reported by: Anonymous
- Originally created at: 2010-08-20T21:11:39.712

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-08-23 14:06:36+00:00_

Should be fixed in <<cset b90ca05941e1fd53e0d69ced376e466dab659f67>> -- there is indeed no need to keep the class objects around and pickle them.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:29:32+00:00_

Removing version: 1.0.1 (automated comment)
